Introduce Header-Stamp in modules CI (#866)
PrestaShop/docs: src/content/1.7/modules/testing/basic-checks.md

Checked all files in 0.085 seconds, 12.000 MB memory used
```

## License headers

When the extension is open-source, sharing the license is one important task.

> [Header-Stamp](https://github.com/PrestaShopCorp/header-stamp) is a tool that applies a header on all compatible files (PHP, JS...).

Reviewer: It should be in the PrestaShop org instead :thinking:

Reviewer 2: This is not something @Quetzacoalt91 can control 😛 .

Author: :grimacing:

Reviewer 2: @PierreRambaud I think we can merge the PR. When the project https://github.com/PrestaShopCorp/header-stamp is moved from PrestaShopCorp to PrestaShop, GitHub will take care of redirecting links?

Author: It usually does, yes. The opposite happened with https://github.com/PrestaShop/ps_checkout, you'll see the redirection is still valid.

In these headers, details can be found about the author, the license, the original year of publication.

> Everytime there is a change to make, one command would apply it to all your files.

Reviewer 3: ```suggestion
One simple command will apply the license header to all your files, or update it as necessary.
```

Author: I would avoid giving an opinion with `simple`. Readers may have another feeling about this.

What about

```suggestion
One command will apply the license header to all your files, or update it as necessary.
```

Reviewer 2: I agree with @Quetzacoalt91 https://www.knowledgeowl.com/home/dont-say-simply-jim-fisher 👍

This tool is part of `prestashop/php-dev-tools` available on [Packagist](https://packagist.org/packages/prestashop/php-dev-tools), which can be required via composer.

> As an example, the license AFL is going to be applied on a PrestaShop native module. There is a AFL template existing in the lib that can be used,

Reviewer 2: ```suggestion
As an example, the license AFL is going to be applied on a PrestaShop native module. There is a AFL template existing in the package that can be used,
```

> and there are a few folder to ignore while these changes are applied (`vendor` because libraries have their own license, test & _dev because they

Reviewer 2: ```suggestion
and there are a few folders to ignore while these changes are applied (`vendor` because libraries have their own license, `test` and `_dev` because they
```

> As an example, the license AFL is going to be applied on a PrestaShop native module. There is a AFL template existing in the package that can be used,
> and there are a few folders to ignore while these changes are applied (`vendor` because libraries have their own license, `test` and `_dev` because they
> won't be part of the release package).

Reviewer 3: I reworded this a bit, then removed the "native" part, because we wouldn't ignore tests and _dev on a native module.
```suggestion
### Example

In the following example, the AFL license is going to be applied on a module. 

The Header-Stamp package already includes an AFL template, so we can use that one. We also have a few folders that we will want to ignore: `vendor` (because libraries have their own license), `test`, and `_dev` (because they won't be part of the release package).
```

> Commands to install and use the tool would be:

Reviewer 3: ```suggestion
To install and use the tool:
```

```bash

> # Install dependencies

Reviewer 3: ```suggestion
# Install header-stamp
```

composer require --dev prestashop/php-dev-tools

# Apply header block
php vendor/bin/header-stamp --license=vendor/prestashop/header-stamp/assets/afl.txt --exclude=vendor,tests,_dev
```

> To return the list of files that would be updated because their header block is missing or different from the one to apply, use the
> `--dry-run` parameter.

Reviewer 3: ```suggestion
To return the list of files that would be updated because their header block is missing or different from the one to apply, use the `--dry-run` parameter:
```
